21 Implement Symptons Form Section

Dylanjb96 opened the pull request:
### Summary
Implemented the Symptoms Form Section for managing multiple symptoms in health record. Part of issue #21.
- Added: 
              - Add/Remove/Update Symptoms.
              - Expand/Collapse Symptoms details.
              - Placeholder on Affected Parts.
              - Display current Symptoms from db.json. 
              - Can Edit/Remove current Symptoms. (Doesn't affect the db.json. Waiting to be use in real database)
              - Feature to require users to fill the field before clicking update Symptoms.
              - CSS for styling and layout.
              Currently on Health Record Page. Waiting for #32  PR to be merged.

### How to Test
1. Run this cmd: npm run dev
2. Ctrl + Click on the localhost uri on terminal.
3. Navigate to the Symptoms section in the Health Record page.
4. Add, edit, and remove symptoms.
5. Ensure that the "Name" and "Start Date" inputs work.
6. Confirm collapsible cards expand/collapse and affected parts placeholder is visible.

### Checklist

- [ ] My code is up-to-date with the `main` branch.
- [ ] My code has no merge conflicts.
- [ ] I have performed a self-review of my code.

danboyce92 reviewed (commented):
Looks good to me, visually it’s very tidy!

I wonder is it possible to get the symptoms to expand smoothly when clicked. I tried messing around with a simple transition rule in the css file but it doesn’t seem to pick it up the way it is put together with the conditional toggle.

One bug I noticed is when I click the add symptom button, the update symptom buttons go faded and my mouse goes under them. Can anyone confirm this on another machine?

Dylanjb96 commented on src/components/SymptomFormSection.css:
```diff
@@ -0,0 +1,120 @@
+  margin-top: 1rem;
+  cursor: pointer;
+}
+
```
I will adjust it.

sttadic commented on src/components/SymptomFormSection.tsx:
```diff
@@ -0,0 +1,120 @@
+        {" "}
+        + Add Symptom
+      </button>
+      {successMessage && <div className="success-message">{successMessage}</div>}
```
This would be better displayed only for symptom that is updated, rather than being a general message

sttadic commented on src/components/SymptomFormSection.tsx:
```diff
@@ -0,0 +1,120 @@
+  const removeSymptom = (index: number) => {
+    const update = symptoms.filter((_, i) => i !== index);
+    setSymptoms(update);
+  };
```
Confirmation for symptom removal could be added here, eg. :
` if (window.confirm("Are you sure you want to remove this symptom?")) {
     // rest of tthe code
  }`

sttadic commented on src/components/SymptomFormSection.tsx:
```diff
@@ -0,0 +1,120 @@
+  };
+
+  const toggleCol = (index: number) => {
+    const update = [...symptoms];
```
This function name is not clear, should be renamed to something like toggleSymptom or toggleSymptomExpansion

sttadic reviewed (commented):
I believe symptom expansion logic should be handled differently, e.g. using state. Nice work though.

Dylanjb96 commented:
@danboyce92 I am unable to solve or find better solution for expanding the symptom smoothly.

Dylanjb96 commented:
- Added: 
              - symtoms management in health record form
              - error icon on cursor hovering on update button when field are not fill
              - removed fetch with data.symptom instead

sttadic reviewed (commented):
To keep things tidy, I’ll consolidate all my feedback into this one main comment, as there’s quite a bit I'd like to address. Some of the comments are more general and relate to the overall form logic and UI, rather then being specific to certain lines of code in this PR.

### Naming inconsistency
There should be more consistency in component and function naming. For example, we have `HealthStatusForm`, `SymptomFormSection` and still pending review `MedicalConsultationsForm`. It would make much more sense to use pattern like [SectionName]Form, e.g. `StatusForm`, `ConsultationsForm`, etc. <br>
Similar issue with functions in `HealthRecordForm`. Some use `handle` prefix while others don't. Example: `removeSymptom` vs. `handleRemoveTreatment`.

### Directory structure
As the codebase continues to grow, the current directory structure is becoming increasingly difficult to navigate. It's starting to be cluttered and harder to quickly understand what belongs where. A reorganization would really help improve clarity and maintainability moving forward (e.g. adding subdirectories).

### HealthRecordForm complexity
This component currently has a lot of visual noise, making it very difficult to read and work with, especially with more logic on the way. It has too many responsibilities in my opinion and extracting some logic into dedicated components like `SymptomsFormSection` would be beneficial for now. We can refactor later introducing helpers, custom hooks, or similar to improve reusability and reduce code duplication. <br> There is also some state complexity. For example, `newTreatment` and `editIndex` for treatment editing, and `symptoms` and `updateSuccess` arrays for symptoms - it's hard to tell at glance what each piece of state is responsible for. My suggestion here would be to remove `updateSuccess` state and logic entirely.

### Treatments section
Why doesn't `Treatments` (or maybe even `Description`) section have its own dedicated component? It's currently embedded, along with all the logic, directly within `HealthRecordForm` adding to overall complexity. <br>
Also, there is inconsistency in how section headings are rendered. For example, `Symptoms` and `Status` use `<h2>`, while `Treatments Tried` and `Description` are using `<label>` tag.

### CSS
Styling approach is starting to become messy, not it terms of visual presentation (which is bad as well :), but in how selectors are being used in specific .css files. For example, in `SymptomsFormSection.css` we have this:
```
button.add-button,
button.remove-button {
  padding: 0.4rem 0.8rem;
  border: none;
  border-radius: 4px;
  background-color: #007bff;
  color: white;
  cursor: pointer;
}

button.add-button {
  background-color: #007bff;
}

button.add-button:hover {
  background-color: #0056b3;
}
```
Using tag based selectors like `button` targets all buttons globally withing the form, not just those in symptoms section. This is actually the third time I'm raising concerns about the styling strategy in this codebase: a lot of repetition, overlapping rules, and collisions. There has to be something we can do, e.g. having only one CSS file for the whole form, and making meaningful and specific class names, and targeting elements properly.<br>
Btw, add symptoms button should be centred.

AladinJmila commented on src/components/SymptomForm.tsx:
```diff
@@ -0,0 +1,68 @@
+  return (
+    <div className="symptom-form-container">
+      <h2>Symptoms</h2>
+      {symptoms.map((symptom, index) => (
```
Since we are rendering a list of symptoms here, it's more accurate to rename this file and the CSS file and have `symptoms` to end with "S" so it will be Symptom`s`Form

AladinJmila commented on src/components/SymptomForm.tsx:
```diff
@@ -0,0 +1,68 @@
+        <div key={index} className="symptom-card">
+          <div className="symptom-header" onClick={() => toggleSymptom(index)}>
+            <h4>Symptom {index + 1}</h4>
+          </div>
```
Could you make a few small tweaks to get the UI looking sleeker and more compact?

First, could you remove the `border-bottom` under the symptom header when it’s collapsed? Actually, let’s just drop it altogether. Also, can you update the `h4` to display the name of the symptom directly, instead of enumerating and repeating the word "Symptom" everywhere? It might not seem like a big deal with just one or two, but if we end up with five symptoms, the repetition won’t look very clean.

When a symptom is expanded, we can hide the `h4` and just show the input fields so the user can edit it.

One more small thing, the Remove button feels a bit too bulky and adds unnecessary weight to the body of the symptom card. Instead, could you replace it with a simple bin icon on the far right of the `h4`? That way, users can delete a symptom without needing to expand it. You can use an icon from the React Icons library that Yuri already installed.

AladinJmila commented on src/components/HealthRecordForm.tsx:
Group all form-related components under a `forms` directory under `components`. Similar to what Stjepan has done with the chat feature, to keep our code easier to navigate. This will probably change in the future, but it's a step in the right direction.

AladinJmila reviewed (changes requested):
We're making some good progress. Let's do a bit more refinment.

AladinJmila commented on src/components/forms/SymptomsForm.css:
I've made the necessary CSS changes directly, one by one, since it was easier than trying to describe them in words. We don’t have a strict style guide to follow, so this way we can move on to the next issue more quickly.
You’ll also notice a few updates in the TSX files.

AladinJmila commented on src/components/forms/SymptomsForm.css:
```diff
@@ -0,0 +1,96 @@
+  max-width: 700px;
+}
+
+.symptom-card {
```
Add this line:
```css
 position: relative;
```

AladinJmila commented on src/components/forms/SymptomsForm.css:
```diff
@@ -0,0 +1,96 @@
+  cursor: pointer;
+  padding: 1rem;
+}
+
```
Replace with:
```css
.symptom-header {
  cursor: pointer;
  text-align: center;
}
```

AladinJmila commented on src/components/forms/SymptomsForm.css:
```diff
@@ -0,0 +1,96 @@
+  font-size: 1rem;
+  margin-left: auto;
+}
+
```
Replace with:
```css
.trash-icon {
  position: absolute;
  top: 0.7rem;
  right: 0.7rem;
  color: #dc3545;
  cursor: pointer;
  font-size: 1.4em;
  padding: 3px;
  border-radius: 4px;
  transition: all 0.2s ease;
}
```

AladinJmila commented on src/components/forms/SymptomsForm.css:
```diff
@@ -0,0 +1,96 @@
+
+.trash-icon:hover {
+  color: #a71d2a;
+}
```
Replace with:
```css
.trash-icon:hover {
  color: white;
  background-color: #dc3545;
}
```

AladinJmila commented on src/components/forms/SymptomsForm.tsx:
```diff
@@ -0,0 +1,67 @@
+              }}
+            />
+          </div>
+
```
Rework the header to match this implementation:
```tsx
<div className="symptom-header" onClick={() => toggleSymptom(index)}>
   <h4>{symptom.name}</h4>
</div>
<FaRegTrashAlt
   className="trash-icon"
   onClick={(e) => {
      e.stopPropagation();
      removeSymptom(index);
   }}
/>
 ```
You’ll notice here that I removed the icon from the "symptom-header" section. It’s better to position it absolutely to the card so it stays in the same place whether the header is collapsed or not, but we can revisit that later. I also removed some redundant classes that were defined but never used in the CSS, so keep an eye out for that going forward. Additionally, I removed the unnecessary div wrapping around the h4.

I also used a different icon—just a personal preference, as it looks a bit less chunky in the UI.

As you'll see now, I kept the name always visible and centered it so it doesn’t clash with the input field. This way, the user can see the name update as they type, and it avoids leaving too much empty space when expanded.

AladinJmila reviewed (approved):
I'm approving this PR now, but please make sure to implement the requested changes exactly as I described. Most of it is copy-paste, but there are a few small changes that might be easy to miss if you try to type it out manually, so I really recommend just copying it to avoid any issues we’d need to fix later.

One other thing to flag, but I’m also not sure if it’s something we should address at this stage. Right now, there’s no validation when adding a new symptom. You can leave both the name and the date empty and still submit, which shouldn’t be possible. We can ignore it for now since you're not handling validation in general, but just be aware that this will need to be revisited later.
